source/common/http/path_utility.cc

```diff
@@ -85,5 +85,63 @@ absl::string_view PathUtil::removeQueryAndFragment(const absl::string_view path)
   return ret;
 }
 
+absl::optional<std::string> PathTransformer::mergeSlashes(absl::string_view original_path) {
+  const absl::string_view::size_type query_start = original_path.find('?');
+  const absl::string_view path = original_path.substr(0, query_start);
+  const absl::string_view query = absl::ClippedSubstr(original_path, query_start);
+  if (path.find("//") == absl::string_view::npos) {
+    return original_path.data();
+  }
+  const absl::string_view path_prefix = absl::StartsWith(path, "/") ? "/" : absl::string_view();
+  const absl::string_view path_suffix = absl::EndsWith(path, "/") ? "/" : absl::string_view();
+  return absl::StrCat(path_prefix, absl::StrJoin(absl::StrSplit(path, '/', absl::SkipEmpty()), "/"),
+                      path_suffix, query);
+}
+
+absl::optional<std::string> PathTransformer::rfcNormalize(absl::string_view original_path) {
+  const auto query_pos = original_path.find('?');
+  auto normalized_path_opt = canonicalizePath(
+      query_pos == original_path.npos
+          ? original_path
+          : absl::string_view(original_path.data(), query_pos) // '?' is not included
+  );
+  if (!normalized_path_opt.has_value()) {
+    return {};
+  }
+  auto& normalized_path = normalized_path_opt.value();
+  const absl::string_view query_suffix =
+      query_pos == original_path.npos
+          ? absl::string_view{}
+          : absl::string_view{original_path.data() + query_pos, original_path.size() - query_pos};
+  if (!query_suffix.empty()) {
+    normalized_path.insert(normalized_path.end(), query_suffix.begin(), query_suffix.end());
+  }
+  return normalized_path;
+}
+
+PathTransformer::PathTransformer(envoy::type::http::v3::PathTransformation path_transformation) {
+  const auto& operations = path_transformation.operations();
```

Comment:
right now we won't reject any configs with dupes -- can you add a comment/TODO?

Reply:
Thanks, I will add the rejection.